### ISSUE TITLE
Symbol argmax API absence

### ISSUE THREAD

**Issue by magic282** [dmlc/mxnet#2171]
Hi,

I am trying to implement the sequence to sequence model using mxnet. It seems that there is no argmax API for mxnet symbol (http://mxnet.readthedocs.io/en/latest/packages/python/symbol.html). So how should I implement the decoder part when emitting a word after softmax? Any suggestions? 

Thanks.

**Comment by sxjscience** [comment]
`argmax` may not be needed for a sequence-to-sequence learning model. During training, we can use softmax + crossentropy to compute the loss. For testing, we can get the output of softmax activation and perform beam-search. 
Also, to add an `argmax` operator, you can fill in the backward implementation of the current `argmax` https://github.com/dmlc/mxnet/blob/master/src/operator/broadcast_reduce_op-inl.h#L183 and register it as a symbol.

**Comment by magic282** [comment]
@sxjscience Thanks for your comment. I will try it!
